datageneration/gpt_data_generator.py

```diff
@@ -397,7 +398,7 @@ def generate_prompts(self, tag_queries: List[LocPoint]) -> List[GeneratedPrompt]
 
         return results
 
-    def generate_sentences(self, generated_prompts) -> List[GeneratedIMRSentence]:
```

Comment:
If I were you I would do writing after executing in generate_sentences, in main block (after line 485). It makes not testable.